tests/test_custom_models.py

```diff
@@ -784,6 +808,51 @@ def forward(self, X):
         return X
 
 
+class ModelConv3D(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv3d = nn.Conv3d(5, 10, 3)
+        self.relu = nn.ReLU()
+        self.flat = nn.Flatten()
+        self.lin0 = nn.Linear(10, 2)
+        self.sm = nn.LogSoftmax(dim=-1)
+
+    def forward(self, X):
+        # If necessary, convert from 2D image to 3D volume
+        if X.dim() == 2:
+            X = torch.stack([X] * 3, dim=-1)
+        X = X.float().reshape(-1, 5, 3, 3, 3)
+        X = self.conv3d(X)
+        X = self.relu(X)
+        X = self.flat(X)
+        X = self.lin0(X)
+        X = self.sm(X)
+        return X
+
+class ModelConv3D2(nn.Module):
```

Comment:
I think this class is not used anywhere?

Reply:
I added some test cases with that in the latest commit

Reply:
Could you explain why we need 2 separate models with `Conv3d`? What does this one cover that the first one does not?

Reply:
I suppose we don't, I was just blindly copying the existing Conv2d. But on closer inspection that one was made specifically for BOFT.

Reply:
Then I would suggest to remove that model, as I think it could be confusing to readers of the code and also makes testing ever so slightly slower.